tokio-threadpool/src/task/mod.rs

```diff
@@ -34,6 +34,21 @@ pub(crate) struct Task {
     /// Next pointer in the queue of tasks pending blocking capacity.
     next_blocking: AtomicPtr<Task>,
 
+    /// ID of the worker that polled this task first.
+    ///
+    /// This field can be a `Cell` because it's only accessed by the worker thread that is
+    /// executing the task.
+    ///
+    /// The worker ID is represented by a `u32` rather than `usize` in order to save some space
+    /// on 64-bit platforms.
+    pub reg_worker: Cell<Option<u32>>,
```

Comment:
Actually, `Task` is `Send` because it can be stolen and moved between worker threads.

What is important here is that `Task` has `Sync` and `!Sync` parts:

* `state` is `Sync` because it may be concurrently written to from multiple threads, e.g. during notification.
* `future` is `!Sync` because it's only ever used by the thread which is currently executing it. `reg_worker` is `!Sync` for the same reason.

It *seems* to me that `reg_worker` can be a `Cell` here just fine, since it's never accessed concurrently. Is my understanding correct? Otherwise, I can switch to `UnsafeCell`, it's no problem.

Reply:
It can be safely called, it was more of a question of how to write idiomatic unsafe rust code :)

Reply:
I'm very interested in the answer to that question, too... this kind of pattern occurs in plenty of places.